### DISCUSSION TITLE
Supported stream formats?

### DISCUSSION THREAD

**Discussion by v-ivanyshyn** [collin80/SavvyCAN#693] [Q&A]
Hi. Is there a reference of which CAN stream formats are supported? I see the list of supported formats in https://github.com/collin80/SavvyCAN/blob/master/README.md, but interested in more details how these formats look. I have a custom logging device that works as serial port. Would like to connect it to SavvyCAN. I see the connection options, so the remaining question is how to customize data stream to let SavvyCAN accept it. If there is a reference/guide how SavvyCAN recognizes the stream format, would be perfect for me.

**Comment by collin80** [comment]
The gvret protocol is documented here: https://github.com/collin80/M2RET/blob/master/CommProtocol.txt

I can't say it's a fabulous protocol. The most permanent solution is the temporary one you made a decade ago... But, it's functional and SavvyCAN supports it. Otherwise you can make your own QT Canbus driver if you want. Or, make a socketcan driver. I wouldn't bother with LAWICEL right now. It's kind of supported but not well enough that you should trust that (yet).

**Reply by v-ivanyshyn** [reply]
Thank you. I also found in the file https://github.com/collin80/SavvyCAN/blob/1999da88c5d6b32bd432b737ba1b60d1a15ff47b/framefileio.cpp that it accepts a very simple format like `43D0029EB2C00000000` which works for me.
